edgedetector/solver/selection/roulette_wheel_selection.py

```diff
@@ -0,0 +1,47 @@
+"""
+This class defines roulette wheel selection operation.
+"""
+from random import random
+
+from edgedetector.solver.selection.Selection import Selection
+
+
+def _get_roulette_table(population):
+    overall_fitness = sum([genotype.fitness for genotype in population])
+    roulette_fitness = {genotype: normalized_fitness for (genotype, normalized_fitness) in zip(
+        [genotype for genotype in population], [genotype.fitness / overall_fitness for genotype in population])}
+    roulette_fitness = sorted(roulette_fitness.items(), key=lambda item: item[1], reverse=True)
+
+    interval_start = 0
+    roulette_table = dict()
+    for (genotype, normalized_fitness) in roulette_fitness:
+        roulette_table[genotype] = (interval_start, interval_start + normalized_fitness)
+        interval_start += normalized_fitness
+
+    return roulette_table
+
+
+def _get_genotype(roulette_table, roulette_result):
+    for (key, value) in roulette_table.items():
+        if value[0] <= roulette_result < value[1]:
+            return key
+
+
+class RouletteWheelSelection(Selection):
+    def select(self, population):
+        roulette_table = _get_roulette_table(population)
+        return self._select_with_repetition(roulette_table)
+
+    def _select_with_repetition(self, roulette_table):
+        return _get_genotype(roulette_table, random()), _get_genotype(roulette_table, random())
+
+    def _select_without_repetition(self, roulette_table):
```

Comment:
maybe it should be configurable which kind of selection we use (with or without repetition)  ?